### PR TITLE
DSD-1718: Modal heading size update

### PR DESCRIPTION
**Pull request by 7emansell**
Fixes JIRA ticket [DSD-1718](https://jira.nypl.org/browse/DSD-1718)

## This PR does the following:
- Adds optional `headingSize` prop to `useDSHeading` hook
- Updates modal default heading size to `heading4`

## How has this been tested?

Locally, Storybook

### Checklist:

- [x] I have updated the Storybook documentation accordingly.
- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

- [x] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

**Review by EdwinGuzman — COMMENTED** [review]
Looks good but just update the global changelog and individual changelog for ~the hook and~ component. Update: the hook is internal so it does not have a changelog.

File: src/components/Modal/modalChangelogData.ts [hunk 1]
```diff
@@ -9,6 +9,13 @@
 import { ChangelogData } from "../../utils/ComponentChangelogTable";
 
 export const changelogData: ChangelogData[] = [
+  {
+    date: "2024-03-26",
```
Comment (EdwinGuzman) [inline src/components/Modal/modalChangelogData.ts]: Make date and version both "Prerelease". This will go out another day.

**Review by EdwinGuzman — APPROVED** [review]
Looks good just minor comment for prerelease date/version.

File: src/components/Modal/modalChangelogData.ts [hunk 2]
```diff
@@ -9,6 +9,13 @@
 import { ChangelogData } from "../../utils/ComponentChangelogTable";
 
 export const changelogData: ChangelogData[] = [
+  {
+    date: "Prerelease",
+    version: "3.0.0",
```
Comment (bigfishdesign13) [inline src/components/Modal/modalChangelogData.ts]: Nit: This should also be set to `Prerelease`.

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,10 @@ Currently, this repo is in Prerelease. When it is released, this project will ad
 
 - Adds the `"editorMode"` option (pencil) to the `Icon` component.
 
+### Updates
+
+- Updates default heading size to `heading4` on Modal component.
```
Comment (bigfishdesign13) [inline CHANGELOG.md:19]: Nit: Also wrap the component name in tick marks. (`Modal`)

**Review by bigfishdesign13 — APPROVED** [review]
The functionality looks good!

**Pull request merged**